### PR TITLE
Add -MOVED and -ASK redirects

### PR DESCRIPTION
**Pull request by zuiderkwast**
From now on, only the instance owning a key (or slot) will reply to GET
and SET requests. The other instances will reply with a redirect.

ASK redirects are implemented as exceptions to the slot mapping. They
are created using fakeredis_cluster:set_ask_redirect/3.

Some function are added to check the slot mapping.

### CODE REVIEW

**Review by bjosv — APPROVED** [review]
nice!

File: src/fakeredis_instance.erl [hunk 1]
```diff
@@ -102,51 +106,111 @@ parse_data(Data, #state{parser_state = ParserState} = State) ->
...
+            ok ->
+                %% Not a redirect. The key is served by us.
+                timer:sleep(State#state.delay),
+                case ets:lookup(?STORAGE, Key) of
+                    [{Key, Val}] ->
+                        Val;
+                    [] ->
+                        null_bulkstring
+                end;
+            {error, _RedirectMsg} = Error ->
```
Comment (bjosv) [inline src/fakeredis_instance.erl:156]: have not looked into this yet, but I started to think how willl {error, xxx} be encoded as..

Comment (bjosv) [reply src/fakeredis_instance.erl:156]: never mind..found out now..

File: src/fakeredis_instance.erl [hunk 2]
```diff
@@ -102,51 +106,111 @@ parse_data(Data, #state{parser_state = ParserState} = State) ->
...
 
 %% Make sure first command is in upper
-handle_data(State, [Cmd | Data]) ->
-    handle_request(State, lists:flatten([string:uppercase(Cmd), Data])).
-
-handle_request(State, [<<"CLUSTER">> | [Type | _Rest]]) ->
+-spec handle_data(Command :: list(), #state{}) -> #state{}.
+handle_data([Cmd | Data], State) ->
+    CmdUpper = string:uppercase(Cmd),
+    handle_request([CmdUpper | Data], State),
```
Comment (bjosv) [inline src/fakeredis_instance.erl:116]: an option might be to have handle_request return a new state, and handle flag setting in the command specific handle_request. but that can be an option for the future if more flags is needed I guess

**Review by bjosv — APPROVED** [review]

File: src/fakeredis_instance.erl (hunk 2 above)
Comment (zuiderkwast) [reply src/fakeredis_instance.erl:116]: Yes, I did that first, but only letting handle_data modify the state is simpler. All the handle_request clauses don't need to think about returning a state then.

**Pull request merged**